src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -786,3 +787,67 @@ export const triggerVirusScanning = (
     return errAsync(new VirusScanFailedError())
   })
 }
+
+/**
+ * Downloads file from clean bucket
+ * @param cleanFileKey object key of the file in the clean bucket
+ * @param versionId id for versioning of the file in the clean bucket
+ * @returns okAsync(buffer) if file has been successfully downloaded from the clean bucket
+ * @returns errAsync(DownloadCleanFileFailedError) if file download failed
+ */
+export const downloadCleanFile = (cleanFileKey: string, versionId: string) => {
+  const logMeta = {
+    action: 'downloadCleanFile',
+    cleanFileKey,
+    versionId,
+  }
+
+  if (!validate(cleanFileKey)) {
+    logger.error({
+      message: 'Invalid clean file key - not a valid uuid',
+      meta: logMeta,
+    })
+
+    return errAsync(new InvalidFileKeyError())
+  }
+
+  let buffer = Buffer.alloc(0)
+
+  const writeStream = new Writable({
+    write(chunk, encoding, callback) {
+      buffer = Buffer.concat([buffer, chunk])
+      callback()
+    },
+  })
+
+  const readStream = AwsConfig.s3
+    .getObject({
+      Bucket: AwsConfig.virusScannerCleanS3Bucket,
+      Key: cleanFileKey,
+      VersionId: versionId,
+    })
+    .createReadStream()
+
+  readStream.pipe(writeStream)
+
+  return ResultAsync.fromPromise(
+    new Promise<Buffer>((resolve, reject) => {
+      readStream.on('end', () => {
+        resolve(buffer)
+      })
+
+      readStream.on('error', (error) => {
+        reject(error)
+      })
+    }),
+    (error) => {
+      logger.error({
+        message: 'Error encountered when downloading file from clean bucket',
+        meta: logMeta,
+        error,
```

Comment:
might need to cast error to String(error) here

Reply:
`error` is typed as `Error` type in `readStream.on` already though, and that's the only path the promise is rejected.